src/main/java/codeu/controller/BotController.java

```diff
@@ -49,9 +51,32 @@ void setUserStore(UserStore userStore) {
 
   /** Adds an entry to the map and to the UserStore. */
   public void registerBot(Bot bot) {
-    botMap.put(bot.getMentionKey().toLowerCase(), bot);
+    // Check if the bot is already in the UserStore
```

Comment:
Are we making sure names are unique while signing up?

Reply:
We only check this if it's all in the same case. I noticed on production we have "user1" and "User1". Perhaps checking if a username contains "bot", ignoring case, can be a quick fix since we really only care about unique bot names.

Reply:
What do you mean by 'checking if a username contains "bot"'? You mean rejecting those usernames on sign up, or for detecting whether a bot with this name has been registered?

Here's another thought. How about using a pinned value for the bot UUIDs like you initially did? Maybe we can hash the bot name and use that as a UUID. When starting up, you'd look up any users whose UUID is equal to this bot's name after applying the same hash function, and that is how you can identify bots.

Reply:
I guess both. My idea is to restrict "bot" as a keyword for actual bots. 

I really like that idea, thanks for suggesting it. So basically if there aren't any matching users with the same UUID, `BotController` can add the bot to the `UserStore`. And if there are, then there would be no need for that.

Reply:
Yes. It's reasonable that the UUID is used for identifying bots, since it's really meant to be a unique field, unlike name.